crates/dojo-core/src/world.cairo

```diff
@@ -334,46 +359,20 @@ mod world {
         fn execute(
             ref self: ContractState, system: felt252, mut calldata: Array<felt252>
         ) -> Span<felt252> {
-            let stack_len = self.call_stack_len.read();
-            self.call_stack.write(stack_len, system);
-            self.call_stack_len.write(stack_len + 1);
-
-            // Get the class hash of the system to be executed
-            let system_class_hash = self.systems.read(system);
-
-            // If this is the initial call, set the origin to the caller
-            let mut call_origin = self.call_origin.read();
-            if stack_len.is_zero() {
-                call_origin = get_caller_address();
-                self.call_origin.write(call_origin);
-            }
+            let system_contract_address = self.systems.read(system);
 
             let ctx = Context {
```

Comment:
I think we can remove ctx entirely

Reply:
We need to track origin. And `ctx.world` seems to provide a familiar way to interact with the world.

Reply:
We need to track origin. And `ctx.world` seems to provide a familiar way to interact with the world.

Reply:
Lemme know what you think @tarrencev